editoast/src/views/rolling_stock.rs

```diff
@@ -189,6 +192,35 @@ pub fn map_diesel_error(e: InternalError, name: impl AsRef<str>) -> InternalErro
     }
 }
 
+impl From<model::Error> for RollingStockError {
+    fn from(e: model::Error) -> Self {
+        match e {
+            model::Error::UniqueViolation { constraint }
+                if constraint == "rolling_stock_name_key" =>
+            {
+                Self::NameAlreadyUsed {
+                    name: String::default(),
+                }
+            }
+            model::Error::CheckViolation { constraint, .. }
+                if constraint == "base_power_class_null_or_non_empty" =>
+            {
+                Self::BasePowerClassEmpty
+            }
+            e => Self::Database(e),
+        }
+    }
+}
+
+impl RollingStockError {
+    fn with_name(self, name: String) -> Self {
```

Comment:
You might be able to avoid this weird function/pattern if `model::Error::UniqueViolation` is able to retain the key that created the conflict.

Reply:
https://github.com/OpenRailAssociation/osrd/pull/10196#discussion_r1905522292